### PR TITLE
Fix VB.NET BC42376: An instance of analyzer SymbolicExecutionRunner cannot be created

### LINKED ISSUE #5436: Fix VB.NET BC42376: An instance of analyzer SymbolicExecutionRunner cannot be created
After updating the "SonarAnalyzer.VisualBasic" package from version 8.35.0.42613 to version 8.36.0.43782, the compiler produces this warning:

> vbc : warning BC42376: An instance of analyzer SonarAnalyzer.Rules.VisualBasic.SymbolicExecutionRunner cannot be created from {my project}\packages\SonarAnalyzer.VisualBasic.8.36.0.43782\analyzers\SonarAnalyzer.VisualBasic.dll : Could not load file or assembly 'SonarAnalyzer.CFG, Version=8.36.0.0, Culture=neutral, PublicKeyToken=c5b62af9de6d7244' or one of its dependencies. The system cannot find the file specified..

Downgrading back to 8.35 eliminates the warning. 

This is in a .NET Framework 4.8 project in Visual Studio 2022 (17.1.0).

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Fixes #5436

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj [hunk 1]
```diff
@@ -50,10 +50,10 @@
 
   <Target Name="CopyBinaries" AfterTargets="Build" DependsOnTargets="SignDlls" Condition=" '$(TargetFramework)' == 'NET46' ">
     <ItemGroup>
+      <BinariesToCopy Include="$(OutputPath)\Google.Protobuf.dll" />
```
Comment (csaba-sagi-sonarsource) [inline analyzers/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj:53]: Why this change? The order in which the binaries are copied should not be important, right? If you just wanted to make sure to have a consistent order for both languages than maybe it is also worth it having the same order of steps for the install and uninstall scripts for both languages...

File: analyzers/packaging/SonarAnalyzer.CSharp.nuspec [hunk 2]
```diff
@@ -22,10 +22,10 @@
     </frameworkAssemblies>
   </metadata>
   <files>
-    <file src="..\src\SonarAnalyzer.CSharp\bin\Release\net46\SonarAnalyzer.CFG.dll" target="analyzers" />
-    <file src="..\src\SonarAnalyzer.CSharp\bin\Release\net46\SonarAnalyzer.dll" target="analyzers" />
-    <file src="..\src\SonarAnalyzer.CSharp\bin\Release\net46\SonarAnalyzer.CSharp.dll" target="analyzers" />
-    <file src="..\src\SonarAnalyzer.CSharp\bin\Release\net46\Google.Protobuf.dll" target="analyzers" />
+    <file src="Binaries\Google.Protobuf.dll" target="analyzers" />
```
Comment (csaba-sagi-sonarsource) [inline analyzers/packaging/SonarAnalyzer.CSharp.nuspec:25]: Could you please tell me where is the Binaries folder coming from? I don't see where is it defined?

File: analyzers/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj (hunk 1 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj:53]: Alphabetical order for readability. Random sort was hard to read/validate.

File: analyzers/packaging/SonarAnalyzer.CSharp.nuspec (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/packaging/SonarAnalyzer.CSharp.nuspec:25]: There's a target in csproj that does this:

https://github.com/SonarSource/sonar-dotnet/blob/39f0d33a9e767b341cdea68a634c6cb2c6ee4d21/analyzers/src/SonarAnalyzer.CSharp/SonarAnalyzer.CSharp.csproj#L51-L59

and

https://github.com/SonarSource/sonar-dotnet/blob/39f0d33a9e767b341cdea68a634c6cb2c6ee4d21/analyzers/src/SonarAnalyzer.VisualBasic/SonarAnalyzer.VisualBasic.csproj#L30-L37

Comment (pavel-mikula-sonarsource) [reply analyzers/packaging/SonarAnalyzer.CSharp.nuspec:25]: Both places were actually updated in this PR 😄 - C# was sorted, VB.NET was sorted and added CFG

Comment (csaba-sagi-sonarsource) [reply analyzers/packaging/SonarAnalyzer.CSharp.nuspec:25]: Yes, and now I found where the folder actually comes from, as it is just used in the `.csproj` files and not defined there.
https://github.com/SonarSource/sonar-dotnet/blob/39f0d33a9e767b341cdea68a634c6cb2c6ee4d21/analyzers/src/Directory.Build.targets#L7

So I think it would be nice to change the casing from Binaries to binaries, for the sake of consistency.

Comment (csaba-sagi-sonarsource) [reply analyzers/packaging/SonarAnalyzer.CSharp.nuspec:25]: Maybe a comment in the Directory.Build.targets mentioning that the path is also used here would ensure that the packaging does not fail if somebody in the future changes the value of the `BinariesFolder`.

File: analyzers/packaging/tools-vbnet/install.ps1 [hunk 3]
```diff
@@ -48,5 +48,8 @@ $project.Object.AnalyzerReferences.Add($analyzerFilePath)
 $analyzerFilePath = Join-Path $analyzersPath "SonarAnalyzer.dll"
 $project.Object.AnalyzerReferences.Add($analyzerFilePath)
 
+$analyzerFilePath = Join-Path $analyzersPath "SonarAnalyzer.CFG.dll"
```
Comment (csaba-sagi-sonarsource) [inline analyzers/packaging/tools-vbnet/install.ps1:51]: NITPICK: WDYT think about making the same order of steps as for the chsarp install script? Theoretically it should ease the update of the scripts if the same action has to be added to both. Same goes for the uninstall script.

File: analyzers/packaging/SonarAnalyzer.CSharp.nuspec (hunk 2 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/packaging/SonarAnalyzer.CSharp.nuspec]: `Directory.Build.targets` is not the right place for such comment, but it makes sense to leave us some note.

It should also fail Java ITs after this PR. And I've created #5442 to add NuGet IT in the future.

Comment (pavel-mikula-sonarsource) [reply analyzers/packaging/SonarAnalyzer.CSharp.nuspec]: If somebody changes `BinariesFolder` in `Directory.Build.targets`, the build will actually fail hard.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
I lefte one Nitpick comment, other than that, looks good!

File: analyzers/packaging/tools-vbnet/install.ps1 (hunk 3 above)
Comment (pavel-mikula-sonarsource) [reply analyzers/packaging/tools-vbnet/install.ps1:51]: It's the CSharp that was wrong. Fixed.
